scripts/PixelmonStaking/staking/rescue.token.js

```diff
@@ -0,0 +1,39 @@
+// We require the Hardhat Runtime Environment explicitly here. This is optional
+// but useful for running the script in a standalone fashion through `node <script>`.
+//
+// When running the script with `npx hardhat run <script>` you'll find the Hardhat
+// Runtime Environment's members available in the global scope.
+const hre = require("hardhat");
+require("dotenv").config();
+
+async function main() {
+    // Hardhat always runs the compile task when running scripts with its command
+    // line interface.
+    //
+    // If this script is run directly using `node` you may want to call compile
+    // manually to make sure everything is compiled
+    // await hre.run('compile');
+
+    const PixelmonStaking = await hre.ethers.getContractFactory("PixelmonStaking");
+    const pixelmonStaking = await PixelmonStaking.attach(process.env.STAKING_CONTRACT);
+
+    const STARTING_TOKEN_ID = Number(process.env.STARTING_TOKEN_ID);
+    const ENDING_TOKEN_ID = Number(process.env.ENDING_TOKEN_ID);
+
+    for(let tokenId = STARTING_TOKEN_ID; tokenId <= ENDING_TOKEN_ID; tokenId++) {
+        console.log(`Rescuing tokenId: ${tokenId}`);
+        let transaction = await pixelmonStaking.rescueNFT(process.env.TRAINER_CONTRACT, process.env.RECEIVER_WALLET, tokenId);
```

Comment:
If the transaction fails in the middle, we have to manually reset the token ids and run again, right?

Reply:
yes. we can see the completed one from log and also from etherscan and update them. Then we can run again